File: libsearchspinner/src/main/java/com/mishaki/libsearchspinner/view/StringSearchSpinner.java
package com.mishaki.libsearchspinner.view;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.Nullable;

/**
 * Created by 杜壁奇<br/>
 * on 2020/08/31 20:31
 */
public class StringSearchSpinner extends SearchSpinner<String> {
    public StringSearchSpinner(Context context) {
        this(context, null, 0);
    }

    public StringSearchSpinner(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public StringSearchSpinner(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setFilterModel((searchContent, s) -> s.contains(searchContent));
        setContentModel(s -> s);
    }
}

File: app/src/main/java/com/mishaki/searchspinner2/view/StringSearchSpinner.kt
package com.mishaki.searchspinner2.view

import android.content.Context
import android.util.AttributeSet
import com.mishaki.searchspinner2.mode.SpinnerContentMode
import com.mishaki.searchspinner2.mode.SpinnerFilterMode

class StringSearchSpinner : SearchSpinner<String> {
    constructor(context: Context) : this(context, null, 0)
    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        filterMode = object : SpinnerFilterMode<String> {
            override fun filterMode(searchContent: String, t: String): Boolean {
                return t.contains(searchContent)
            }
        }
        contentMode = object : SpinnerContentMode<String> {
            override fun getContent(t: String): String {
                return t
            }
        }
    }
}